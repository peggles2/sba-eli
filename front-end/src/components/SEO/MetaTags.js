import React, { Component } from "react";
import { Helmet } from "react-helmet";

export default class MetaTags extends Component {
  getMetaTitle(pageTitle) {
    return pageTitle
      ? pageTitle
      : "Small Business Administration: Women Entrepreneurs";
  }

  getMetaDescription(description) {
    return description
      ? description
      : "We support America's small businesses. The SBA connects entrepreneurs with lenders and funding to help them plan, start and grow their business.";
  }

  getCanonicalUrl(url) {
    return url && url.startsWith("https://") ? url : "https://sba.gov/eli";
  }

  getMetaImage(imageUrl) {
    return imageUrl ? imageUrl : "https://picsum.photos/200/300/?random";
  }

  render() {
    const metaProps = [
      {
        name: "description",
        content: this.getMetaDescription(this.props.metaDescription)
      },
      {
        property: "og:title",
        content: this.getMetaTitle(this.props.metaTitle)
      },
      {
        property: "og:description",
        content: this.getMetaDescription(this.props.metaDescription)
      },
      {
        property: "og:type",
        content: "website"
      },
      {
        property: "og:image",
        content: this.getMetaImage(this.props.metaImage)
      },
      {
        property: "og:url",
        content: this.getCanonicalUrl(this.props.canonicalUrl)
      }
    ];

    const linkProp = {
      rel: "canonical",
      href: this.getCanonicalUrl(this.props.canonicalUrl)
    };
    return (
      <Helmet
        title={this.getMetaTitle(this.props.metaTitle)}
        meta={metaProps}
        link={linkProp}
      />
    );
  }
}
