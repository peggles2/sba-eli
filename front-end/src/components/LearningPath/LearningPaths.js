import React from 'react';
import LearningPathsList from './LearningPathsList';
import LearningPathAbout from './LearningPathAbout';
import LearningPathResume from './LearningPathResume';
import MetaTags from '../SEO/MetaTags'
import { withRouter } from 'react-router-dom';
import { Container, Divider, Grid, Header } from 'semantic-ui-react';
import { connect } from "react-redux";

import { toggleModal } from '../../actions/aboutModalActions'

export class LearningPaths extends React.Component {
  learningPathResume = (isUserLoggedIn) => {
    if(isUserLoggedIn) {
      return(
          <Grid.Row>
            <Grid.Column>
              <LearningPathResume />
            </Grid.Column>
          </Grid.Row>
      );
    };
  };

  render () {
    const isUserLoggedIn = this.props.isUserLoggedIn;

    return(
      <div>
        <MetaTags metaTitle="SBA Learning Paths"
                metaDescription="Description for the Learning Path landing page"
                canonicalUrl="https://sba.gov/learning_paths"/>
        <Grid>
          {this.learningPathResume(isUserLoggedIn)}
          <Divider hidden />
          <Grid.Row>
            <Grid.Column>
              <Container>
                <Header as='h2'>{isUserLoggedIn ? "Other Journeys you have Explored" : "Explore All Journeys"}</Header>
                <LearningPathAbout open={this.props.displayModal} handleClose={this.props.handleModalClose} />
                <Divider />
              </Container>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column>
              <Container>
                <LearningPathsList />
              </Container>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </div>
    )
  };
};

const mapStateToProps = store => {
  return {
    displayModal: store.aboutModal.show,
    isUserLoggedIn: store.login.isUserLoggedIn,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    handleModalClose: () => { dispatch(toggleModal(false)) },
  };
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(LearningPaths));
