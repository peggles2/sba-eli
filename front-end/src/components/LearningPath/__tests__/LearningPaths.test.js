import React from 'react';
import { LearningPaths } from '../LearningPaths';
import MetaTags from '../../SEO/MetaTags';
import { shallow, mount } from 'enzyme';

describe('LearningPaths', () => {
  it("should render meta tags with Learning Paths specific information", () => {
    const wrapper = shallow(<LearningPaths />);

    expect(wrapper.find(MetaTags).exists()).toBe(true);
    const metaWrapper = wrapper.find(MetaTags).dive();

    expect(metaWrapper.find('title').text()).toBe("SBA Learning Paths");
    expect(metaWrapper.find("meta[name='author']").exists()).toBe(false);
    expect(metaWrapper.find("meta[name='description']").exists()).toBe(true);
    expect(metaWrapper.find("meta[name='description']").props().content).toBe("Description for the Learning Path landing page");
    //TODO: add Canonical URL validation when we have final urls
  });

  it('should render a <div>', () => {
    const wrapper = shallow(<LearningPaths />);

    expect(wrapper.find('div').length).toEqual(1);
  });
});
