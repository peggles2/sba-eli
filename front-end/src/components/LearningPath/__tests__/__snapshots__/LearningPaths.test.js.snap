// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningPaths should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningPaths />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Learning Paths
        </h2>,
        <LearningPathsList />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Learning Paths",
        },
        "ref": null,
        "rendered": "Learning Paths",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Learning Paths
          </h2>,
          <LearningPathsList />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Learning Paths",
          },
          "ref": null,
          "rendered": "Learning Paths",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
